plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/metastore/HiveMetastoreBackedDeltaLakeMetastore.java

```diff
@@ -118,7 +157,9 @@ public HiveMetastoreBackedDeltaLakeMetastore(
     {
         Optional<Table> candidate = delegate.getTable(databaseName, tableName);
         candidate.ifPresent(table -> {
-            if (!TABLE_PROVIDER_VALUE.equalsIgnoreCase(table.getParameters().get(TABLE_PROVIDER_PROPERTY))) {
+            if (((table.getTableType().equals(EXTERNAL_TABLE.name()) || table.getTableType().equals(MANAGED_TABLE.name())) &&
```

Comment:
one check per line pls (for readability sake)

Reply:
Might be better to split these checks into two static util methods so that this line reads `!isDeltaDataTable(table) && !isDeltaView(table)`